test/jdk/java/nio/file/Files/CreateDirectories.java

```diff
@@ -0,0 +1,123 @@
+ * @library ..
+ * @run testng CreateDirectories
+ */
+public class CreateDirectories {
+
+    /**
+     * Creates a symlink which points to a directory that exists. Then calls Files.createDirectories
+     * method passing it the path to the symlink and verifies that no exception gets thrown
+     */
+    @Test
+    public void testSymlinkDir() throws Exception {
+        // create a temp dir as the "root" in which we will run our tests.
+        final Path startingDir = Files.createTempDirectory(Path.of("."), "8293792");
+        if (!TestUtil.supportsLinks(startingDir)) {
+            System.out.println("Skipping tests since symbolic links isn't " +
+                    "supported under directory "+ startingDir);
+            throw new SkipException("Symbolic links not supported");
+        }
+        System.out.println("Running tests under directory " + startingDir.toAbsolutePath());
+        final Path fooDir = Files.createDirectory(Path.of(startingDir.toString(), "foo"));
+        assertTrue(Files.isDirectory(fooDir),
+                fooDir + " was expected to be a directory but wasn't");
+
+        // now create a symlink to the "foo" dir
+        final Path symlink = Files.createSymbolicLink(
+                Path.of(startingDir.toString(), "symlinkToFoo"), fooDir.toAbsolutePath());
+        assertTrue(Files.isSymbolicLink(symlink),
+                symlink + " was expected to be a symlink but wasn't");
+        assertTrue(Files.isDirectory(symlink),
+                symlink + " was expected to be a directory but wasn't");
+
+        // now create a directory under the symlink (which effectively creates a directory under
+        // "foo")
+        final Path barDir = Files.createDirectory(Path.of(symlink.toString(), "bar"));
+        assertTrue(Files.isDirectory(barDir),
+                barDir + " was expected to be a directory but wasn't");
+        // ultimately, we now have this directory structure:
+        // <root-dir>
+        //   |--- foo
+        //   |     |--- bar
+        //   |
+        //   |--- symlinkToFoo -> (links to) <absolute-path-to-root-dir>/foo
+
+
+        // now call Files.createDirectories on each of these existing directory/symlink paths
+        // and expect each one to succeed
+        Files.createDirectories(fooDir); // ./<root-dir>/foo
+        Files.createDirectories(symlink); // ./<root-dir>/symlinkToFoo
+        Files.createDirectories(barDir); // ./<root-dir>/symlinkToFoo/bar
+    }
+
+    /**
+     * Tests Files.createDirectories
+     */
+    @Test
+    public void testCreateDirectories() throws IOException {
+        final Path tmpdir = TestUtil.createTemporaryDirectory();
+        // a no-op
+        Files.createDirectories(tmpdir);
+
+        // create one directory
+        Path subdir = tmpdir.resolve("a");
+        Files.createDirectories(subdir);
+        assertTrue(Files.exists(subdir), subdir + " was expected to exist, but didn't");
```

Comment:
I realise this is moved from Misc.java but if you replace the "exists" with "isDirectory" then it improve the testing.